src/adagrams.js

```diff
@@ -1,15 +1,84 @@
+import {
+  LETTER_POOL, 
+  SCORE_MAP
+} from "../src/data.js";
+
 export const drawLetters = () => {
   // Implement this method for wave 1
+  const drawPool = [];
+  for (const [key, value] of Object.entries(LETTER_POOL)) {
+    for (let i = 0; i < value; i++) {
+      drawPool.push(key);
+    }
+  }
+  const hand = []; 
+  for (let i = 0; i < 10; i++) {
+    const pieceIndex = Math.floor(Math.random() * drawPool.length);
+    const piece = drawPool[pieceIndex];
+    hand.push(piece);
+    const index = drawPool.indexOf(piece);
+    if (index > -1) {
+      drawPool.splice(index, 1);
+    }
+  }
+  return hand;
+};
+
+const letterFreq = (string) => {
+  const freqMap = {};
+  for (const letter of string) {
+    freqMap[letter] = freqMap[letter] ? freqMap[letter] + 1 : 1;
+  }
+  return freqMap;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const uppercaseInput = input.toUpperCase();
+  const splitInput = uppercaseInput.split("");
+  const inputFreq = letterFreq(splitInput);
+  const handFreq = letterFreq(lettersInHand);
+
+  for (const letter of splitInput) {
+    if (!handFreq[letter]) {
+      return false;
+    } else if (inputFreq[letter] > handFreq[letter]) {
+      return false;
+    } 
+  }
+  return true;
 };
 
 export const scoreWord = (word) => {
   // Implement this method for wave 3
+  let points = 0;
+  word = word.toUpperCase();
+  const newWord = word.split("");
+  for (const letter of newWord) {
+    points += SCORE_MAP[letter];
+  } if (newWord.length >= 7) {
+    points += 8;
+  }    
+  return points;
 };
 
 export const highestScoreFrom = (words) => {
   // Implement this method for wave 4
+  let highScore = ['', 0];
+  for (let i = 0; i < words.length; i++) {
+    if (scoreWord(words[i]) > highScore[1]) {
+      highScore[1] = scoreWord(words[i]);
+      highScore[0] = words[i];
+    } else if (scoreWord(words[i]) === highScore[1]) {
+      let highWord = highScore[0]; {
+      if (highWord.length === 10) {continue; } 
```

Comment:
While js doesn't care about white spaces, putting `continue` on the next line is easier to read